README.md

```diff
@@ -108,6 +108,34 @@ getAllDatasets().then(({ jsonld, errors }) => {
 });
 ```
 
+### `validateJsonLdId(id, expectHtml)`
+
+#### Description
+This function validates the `@id` (or `id`, for backwards compatibility) property within a JSON-LD `Dataset` or `DataCatalog`. It fetches JSON-LD data from a specified URL, checks whether the data is embedded in HTML or raw JSON-LD, extracts the JSON-LD, and ensures that the `@id` field within the document matches the provided `id`. This function acts as a safety check, affirming that the expected identifier aligns exactly with the identifier found within the fetched JSON-LD document. Note that `@id` is case sensitive and must match exactly.
+
+#### Parameters
+- `id` (string): A string that specifies the expected `@id` or `id` value in the JSON-LD document.
+- `expectHtml` (boolean): A boolean flag indicating whether the fetched data is expected to be embedded within HTML (when `true`) or expected to be raw JSON-LD (when `false`).
```

Comment:
Why must this function also specify whether the document is JSON-LD vs HTML? It seems like the function could figure it out itself.

Do you have a use case in mind where users need to validate both the ID and the document type simultaneously? It seems like it would be simpler to seperate these concerns

Reply:
Ah I see now that the context isn't clear - Data Catalogue files are JSON-LD, where as Dataset Sites are HTML, and as this is validating it asserts one or the other depending on the context